internal/services/apicenter/apicenter_environment_resource.go

```diff
@@ -0,0 +1,251 @@
+	return environments.ValidateEnvironmentID
+}
+
+func (r ApiCenterEnvironmentResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			service, err := services.ParseServiceID(model.ServiceId)
+			if err != nil {
+				return fmt.Errorf("parsing ApiCenter Service ID %s: %+v", model.ServiceId, err)
+			}
+
+			// @favoretti: can't find any workspace creation buttons in the portal, assume everything defaults to "default" for now?
+			id := environments.NewEnvironmentID(subscriptionId, service.ResourceGroupName, service.ServiceName, "default", model.Identification)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing ApiCenter Environment %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			apiCenterEnvironmentProps := environments.EnvironmentProperties{
+				Kind:        environments.EnvironmentKind(model.Type),
+				Description: &model.Description,
+				Onboarding: &environments.Onboarding{
+					DeveloperPortalUri: &[]string{model.DevPortalUri},
+					Instructions:       &model.Instructions,
+				},
+				Server: &environments.EnvironmentServer{
+					Type:                pointer.To(environments.EnvironmentServerType(model.ServerType)),
+					ManagementPortalUri: &[]string{model.MgmtPortalUri},
+				},
+				Title: model.Name,
+			}
+
+			apiCenterEnvironment := environments.Environment{
+				Name:       &model.Identification,
+				Properties: &apiCenterEnvironmentProps,
+				Type:       &model.Type,
+			}
+
+			if _, err = client.CreateOrUpdate(ctx, id, apiCenterEnvironment); err != nil {
+				return fmt.Errorf("creating ApiCenter Environment %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ApiCenterEnvironmentResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.EnvironmentsClient
+			id, err := environments.ParseEnvironmentID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state ApiCenterEnvironmentResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("reading ApiCenter Environment %s: %v", id, err)
+			}
+
```

Comment:
It looks like we're missing the part where we check if a property has changes then patching it into the existing model before updating the resource?